services/fileuploader/fileuploader.go

```diff
@@ -99,6 +99,10 @@ func (p *provider) updateLoop(ctx context.Context, backendConfig backendconfig.B
 				bucket = getDefaultBucket(ctx, config.GetString("JOBS_BACKUP_STORAGE_PROVIDER", "S3"))
 			}
 			preferences = c.Settings.DataRetention.StoragePreferences
+			// If bucket type is empty string or configuration is empty, clearing preferences so that no backup happens for that workspace.
+			if bucket.Type == "" || len(bucket.Config) == 0 {
+				preferences = backendconfig.StoragePreferences{}
```

Comment:
You're saying `preferences should be false` but `preferences` is not a `boolean`. What is the side effect of having an empty `StoragePreferences` down in the workflow?

Reply:
Shouldn't we not set the map key and delete it if present? So that `settings[c.WorkspaceID]` won't exist.

This way `GetStoragePreferences` would return the proper error:

```go
	if !ok {
		return prefs, fmt.Errorf("no storage settings found for workspace: %s", workspaceID)
	}
```

Reply:
> You're saying `preferences should be false` but `preferences` is not a `boolean`. What is the side effect of having an empty `StoragePreferences` down in the workflow?

By preferences being false, I meant all the boolean inside are false. When we have StoragePreferences empty - all of the dumps will be set as false by default so we will not backup them

Reply:
> Shouldn't we not set the map key and delete it if present? So that `settings[c.WorkspaceID]` won't exist?
> 
> This way `GetStoragePreferences` would return the proper error:
> 
> ```go
> 	if !ok {
> 		return prefs, fmt.Errorf("no storage settings found for workspace: %s", workspaceID)
> 	}
> ```

We are not deleting them, we just assume that preferences by the customer is no backup (all set to false) if we are given an invalid bucket